crates/bevy_ecs/src/system/mod.rs

```diff
@@ -276,6 +276,60 @@ mod tests {
         assert_eq!(*world.resource::<SystemRan>(), SystemRan::Yes);
     }
 
+    #[test]
+    fn get_many_is_ordered() {
+        use crate::system::Resource;
+        const ENTITIES_COUNT: usize = 1000;
+
+        #[derive(Resource)]
+        struct EntitiesArray(Vec<Entity>);
+
+        fn query_system(
+            mut ran: ResMut<SystemRan>,
+            entities_array: Res<EntitiesArray>,
+            q: Query<&W<usize>>,
+        ) {
+            let entities_array: [Entity; ENTITIES_COUNT] =
+                entities_array.0.clone().try_into().unwrap();
+
+            for (i, w) in (0..ENTITIES_COUNT).zip(q.get_many(entities_array).unwrap()) {
+                assert_eq!(i, w.0);
+            }
+
+            *ran = SystemRan::Yes;
+        }
+
+        fn query_system_mut(
+            mut ran: ResMut<SystemRan>,
+            entities_array: Res<EntitiesArray>,
+            mut q: Query<&mut W<usize>>,
+        ) {
+            let entities_array: [Entity; ENTITIES_COUNT] =
+                entities_array.0.clone().try_into().unwrap();
+
+            #[allow(unused_mut)]
+            for (i, mut w) in (0..ENTITIES_COUNT).zip(q.get_many_mut(entities_array).unwrap()) {
+                assert_eq!(i, w.0);
+            }
+
+            *ran = SystemRan::Yes;
+        }
+
+        let mut world = World::default();
+        world.insert_resource(SystemRan::No);
+        let entity_ids = (0..ENTITIES_COUNT)
+            .map(|i| world.spawn(W(i)).id())
+            .collect();
+        world.insert_resource(EntitiesArray(entity_ids));
+
+        run_system(&mut world, query_system);
+        assert_eq!(*world.resource::<SystemRan>(), SystemRan::Yes);
```

Comment:
Is this needed? The executor will propagate the panic up to the test's thread, which should panic and fail the test if the assert is incorrect.

Reply:
This is needed to verify that the system ran at all. This is similar to the way the other tests are written in this module.